Realm/Realm.Weaver/Analytics/AnalyticsUtils.cs

```diff
@@ -190,6 +169,64 @@ public static string GetLanguageVersion(string targetFramework)
             return Unknown();
         }
 
+        // Knowledge on unique machine Ids for different OSes obtained from https://github.com/denisbrodbeck/machineid
+        public static string GetAnonymizedUserId()
+        {
+            var id = string.Empty;
+            var currentOs = System.Environment.OSVersion.Platform;
+            Match match = null;
+            try
+            {
+                if (currentOs == PlatformID.Win32S || currentOs == PlatformID.Win32Windows ||
+                    currentOs == PlatformID.Win32NT || currentOs == PlatformID.WinCE)
+                {
+                    var machineIdToParse = RunProcess("reg", "QUERY HKEY_LOCAL_MACHINE\\SOFTWARE\\Microsoft\\Cryptography -v MachineGuid");
+                    var regex = new Regex("^\\s+MachineGuid\\s+\\w+\\s+((\\w+-?)+)", RegexOptions.Multiline);
+                    match = regex.Match(machineIdToParse);
+                }
+                else if (currentOs == PlatformID.MacOSX)
+                {
+                    var machineIdToParse = RunProcess("ioreg", "-rd1 -c IOPlatformExpertDevice");
+                    var regex = new Regex("^.*IOPlatformUUID\\\"\\s=\\s\\\"(.+)\\\"", RegexOptions.Multiline);
+                    match = regex.Match(machineIdToParse);
+                }
+                else if (currentOs == PlatformID.Unix)
+                {
+                    try
+                    {
+                        id = File.ReadAllText("/etc/machine-id");
+                    }
+                    catch
+                    {
+                        id = string.Empty;
+                    }
```

Comment:
Why do we need this try-catch?

Reply:
For extra security. I know Lubo keeps saying "the file will always be there". But why crashing for a file not found for analytics?

Reply:
I agree that the metrics must never tear down the build for a customer.

In dart (to catch errors in the metrics) we do run the metrics if REALM_CI is set, but without uploading and also without the error suppression. This is to catch errors in the metrics that would otherwise go unnoticed since we normally always suppress errors, and don't even run them if we detect we are on CI.
```dart
  final skipUpload = isRealmCI || Platform.environment['CI'] != null || Platform.environment['REALM_DISABLE_ANALYTICS'] != null;
  if (skipUpload && !isRealmCI) { // skip early, unless REALM_CI is set
    // skip early
    log.info('Skipping metrics upload');
    return;
  }

// calculate metrics

  if (skipUpload) {
    // skip late
    log.info('Skipping metrics upload (late)');
    return;
  }

// do actual upload
```

Reply:
Yeah, my point was that we have an outer try-catch wrapping everything and returning `unknown`, so the inner try catch doesn't seem particularly useful.